Examples/CreatingCaptureFile/Program.cs

```diff
@@ -85,23 +85,25 @@ public static void Main()
         /// <summary>
         /// Prints the time and length of each received packet
         /// </summary>
-        private static void device_OnPacketArrival(object sender, CaptureEventArgs e)
```

Comment:
Would it be easier to make `CaptureEventArgs` a ref struct?
a ref struct may have a Span as field.

See https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct#ref-struct

Reply:
Another way to do this, that is more backward compatible is to:
* Store the data pointer as private field in  `CaptureEventArgs`
* Use propery getters in `CaptureEventArgs` to create a Span or RawCapture

Reply:
@kayoub5 I don't see a path where we create the Span<> as a property getter since it doesn't appear that there is a way to return a Span<>. I tried to find a case where that could work but don't see a way to do that.

Now if we put the Span<> into CaptureEventArgs we could also put an accessor for a RawPacket that converts the Span<> and header to the RawPacket. That should work since the Span<> is retained in the context of the callback (as a part of the CaptureEventArgs reference).

Reply:
@kayoub5 alright, pushed up using CaptureEventArgs getter which looks much cleaner btw, it may actually be entirely transparent to users in fact. Mind taking another look?